src/api/hypixel.ts

```diff
@@ -100,40 +106,52 @@ export async function getPlayer(uuid: string): Promise<Record<any, any>> {
 	return player;
 }
 
-export async function getProfile(uuid: string): Promise<Record<string, any>> {
-	if (isUUID(uuid) === false) {
-		const playerUUID = await getUUID(uuid);
-		if (playerUUID === null) {
-			throw new Error('Player not found!');
-		}
-
-		uuid = playerUUID;
+export async function getProfiles(paramPlayer: string, paramProfile?: string): Promise<GetProfiles> {
+	const uuid = await getUUID(paramPlayer);
+	if (uuid === null) {
+		throw new Error('Player not found!');
 	}
 
 	const storedProfiles = await MONGO.collection<StoredSkyblockProfile>('profiles')
 		.find({ [`profile.members.${uuid}`]: { $exists: true } })
 		.toArray();
-	if (storedProfiles.length && storedProfiles.every((profile) => profile.lastUpdated + profileCacheTTL > Date.now())) {
-		console.log('Returning cached profile data');
-		return storedProfiles.map((profile) => profile.profile);
-	}
 
-	const profiles = await hypixelRequest({ endpoint: 'skyblock/profiles', query: { uuid }, usesApiKey: true })
-		.then((res) => res.profiles)
-		.catch((err) => {
-			throw new Error(err);
-		});
+	let profiles = storedProfiles.map((profile) => profile.profile);
+	if (
+		!storedProfiles.length &&
```

Comment:
This check would mean that new profiles are only fetched if there's already existing profiles

`storedProfiles.length`: Count of stored profiles for this player
`!storedProfiles.length`: True if there are no stored profiles

So `if (!storedProfiles.length && ...)` only passes the check if there aren't any profiles already which is not what we want